packages/firestore/src/local/shared_client_state_schema.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BatchId, MutationBatchState, TargetId } from '../core/types';
+import { QueryTargetState } from './shared_client_state_syncer';
+import { assert } from '../util/assert';
+import { ClientId } from './shared_client_state';
+import { User } from '../auth/user';
+
+// The format of the LocalStorage key that stores the client state is:
+//     firestore_clients_<persistence_prefix>_<instance_key>
+export const CLIENT_STATE_KEY_PREFIX = 'firestore_clients';
+
+/** Assembles the key for a client state in WebStorage */
+export function createWebStorageClientStateKey(
```

Comment:
Out of curiosity (I don't care too much), does the fact that these functions are defined in this .d.ts file mean that they won't get minified?  Or is that not how it works?

Reply:
The function name doesn't, but I thought it would be cleaner to have all schema-related information in this one file.